addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java

```diff
@@ -311,6 +312,20 @@ public AutomationPlan loadPlan(InputStream in) throws AutomationJobException {
         return new AutomationPlan(env, jobsToRun, progress);
     }
 
+    public void loadAndRunPlanPanel(AutomationPlan plan, boolean setFocus, boolean run) {
+        getAutomationPanel().loadPlan(plan);
+        if (setFocus) {
+            getAutomationPanel().setTabFocus();
+        }
+        if (run) {
+            getAutomationPanel().runPlan(plan);
+        }
+    }
+
+    public AutomationJob getJobByEvent(Event e) {
```

Comment:
This could fail if there is no GUI. It should either check `hasView()` or use another mechanism